### ISSUE TITLE
chroot() issues

### ISSUE THREAD

**Issue by wablair** [unbit/uwsgi#1748]
Hello,

I am working in OpenBSD 6.2 with OpenBSD's httpd, uWSGI 2.0.16, Python 3.6, and Django 2.0.2 and attempting to use chroot but it does not seem to be working.

The ini I am using and the logfile can be found at: https://gist.github.com/wablair/83d5538e201d445430db3b3bd35892e7

It looks like uWSGI is trying to access Python home with the full path instead of a shortened path for use in the chroot. But when I remove /var/www from the home variable in the ini I get the following error:

!!! Python Home is not a directory: /env !!!
Set PythonHome to /env
Fatal Python error: Py_Initialize: Unable to get the locale encoding
ModuleNotFoundError: No module named 'encodings'

If I drop to chroot() in core/uwsgi.c in uwsgi_setup() and remove the chroot() calls in core/utils.c and trim the /var/www from paths in the ini for everything except daemonize, it seems to work fine.

It might be possible that Py_SetPythonHome() could be called by the spawned workers with a corrected path but I am still looking into that.

Any pointers on possible configuration issues or where to look is appreciated. Thanks.

**Comment by DanAtShenTech** [comment]
@wablair, have you solved this problem yet?  I've encountered the same issue and have posted a [StackOverflow question](https://stackoverflow.com/q/49144225/5548239) about it.  I also found [this Github issue](https://github.com/unbit/uwsgi/issues/1688) on the uWSGI Github page and have posted to it as well.

**Comment by wablair** [comment]
@DanAtShenTech No, I was hoping an uwsgi contributor would chime in with how/when they would prefer to drop to chroot and handle the change in paths.

**Comment by DanAtShenTech** [comment]
@wablair I've updated my [Stackoverflow question](https://stackoverflow.com/questions/49144225/uwsgi-will-not-run-in-mixed-python-environment-in-order-to-operate-correctly-wit) with [how I worked around this problem](https://stackoverflow.com/a/49162839/5548239).

**Comment by DanAtShenTech** [comment]
I just updated my [StackOverflow](https://stackoverflow.com/questions/49144225/uwsgi-will-not-run-in-mixed-python-environment-in-order-to-operate-correctly-wit) question with another workaround. My first post about a workaround (above) turned out to only work on the command line. I'm still looking for a real solution.
